sdk/core/core-https/src/nodeHttpsClient.ts

```diff
@@ -144,14 +145,14 @@ export class NodeHttpsClient implements HttpsClient {
             reject(new AbortError("The operation was aborted."));
           }
         });
-        if (body) {
-          if (isReadableStream(body)) {
-            body.pipe(req);
-          } else {
-            req.write(body);
-          }
+        if (body && isReadableStream(body)) {
```

Comment:
Is this intended to be a refactoring or behavior change?  It looks like a slight behavior change in some of the cases.

* `body && isReadableStream(body)`
  * Before: `body.pipe(req); req.end()`
  * After: `body.pipe(req);`
* `body && !isReadableStream(body)`
  * Before: `req.write(body); req.end()`
  * After: `req.end(body);`
  * Are these equivalent?
* `!body`
  * No change

Reply:
Yes, this is intentional. What I didn't know when I wrote the code originally was that `pipe()` will automatically call `end()` when the readable emits end:

> By default, stream.end() is called on the destination Writable stream when the source Readable stream emits 'end', so that the destination is no longer writable.  [Source](https://nodejs.org/api/stream.html#stream_readable_pipe_destination_options)

For the other case, you can call `end()` directly with the last chunk, no need to `write()` and then `end()`. We even do this above with the other streams.